### PR TITLE
Upgrade to latest shred version

### PR DESCRIPTION
**Pull request by torkleyy**
This change is [Reviewable](https://reviewable.io/reviews/slide-rs/specs/369)

### CODE REVIEW

File: src/storage/mod.rs [hunk 1]
```diff
@@ -499,8 +515,32 @@ where
...
+/// if there's no default.
+pub trait TryDefault: Sized {
+    /// Tries to create the default.
+    fn try_default() -> Result<Self, String>;
+
+    /// Calls `try_default` and panics on an error case.
+    fn unwrap_default() -> Self {
+        match Self::try_default() {
+            Ok(x) => x,
+            Err(e) => panic!("Failed to create a default value for storage ({:?})", e),
```
Comment (WaDelma) [inline src/storage/mod.rs]: This could be `expect`

File: src/world/mod.rs [hunk 2]
```diff
@@ -206,18 +205,32 @@ impl World {
     where
         T::Storage: Default,
     {
-        self.register_with_storage::<T>(Default::default());
+        self.register_with_storage::<_, T>(|| Default::default());
```
Comment (WaDelma) [inline src/world/mod.rs]: s/|| Default::default()/Default::default/

**Comment by torkleyy** [comment]
bors delegate=Rhuagh

**Pull request merged**